src/Illuminate/Hashing/ArgonHasher.php

```diff
@@ -118,6 +118,56 @@ public function needsRehash($hashedValue, array $options = [])
         ]);
     }
 
+    /**
+     * Verifies that the configuration is less than or equal to what is configured.
+     *
+     * @internal
+     */
+    public function verifyConfiguration($value)
+    {
+        return $this->verifyAlgorithm($value) && $this->verifyOptions($value);
+    }
+
+    /**
+     * Verify the hashed value's algorithm.
+     *
+     * @param  string  $hashedValue
+     * @return bool
+     */
+    protected function verifyAlgorithm($hashedValue)
+    {
+        return $this->info($hashedValue)['algoName'] === 'argon2i';
+    }
+
+    /**
+     * Verify the hashed value's options.
+     *
+     * @param  string  $hashedValue
+     * @return bool
+     */
+    protected function verifyOptions($hashedValue)
+    {
+        ['options' => $options] = $this->info($hashedValue);
+
+        if (
+            ! is_int($options['memory_cost'] ?? null) ||
+            ! is_int($options['time_cost'] ?? null) ||
+            ! is_int($options['threads'] ?? null)
+        ) {
+            return false;
+        }
+
+        if (
+            $options['memory_cost'] > $this->memory ||
+            $options['time_cost'] > $this->time ||
+            $options['threads'] > $this->threads
```

Comment:
I think we should compare the other way, allowing more threads (maybe until a limit) because less threads means more computation time